### PR TITLE
Apply linting

### PR DESCRIPTION
**Pull request by jaguillette**
This PR applies code style and linting checks to the MM API codebase. Specifically, it applies `flake8`, `black`, and `isort`. It has github actions configured to check that code conforms to these standards, and also sets up pre-commit hooks that can be installed to check on the code as part of committing to the repo.

It also implements `poetry` for dependency management instead of relying on `requirements.txt`. The benefit is that it's easier to keep your local environment in sync with dependencies as articulated in pyproject.toml with a `poetry add`. It also creates a `poetry.lock` file that specifies the exact dependencies as installed in one environment, so that the exact same dependencies can be easily installed elsewhere. It does add some complexity to the docker build and slows it down, but it's been helpful in local development, so I want to try it out in a shared project.

The README has been updated with an overview of setting up with poetry, so please let me know if that overview needs improvement.

The code standards have been applied to the codebase, but there should be no functional changes to the app. All of the substantive changes are outside of the `media_management_api` directory, except for `manage.py`, which also got some non-functional style changes. The `poetry.lock` file is generated programmatically, so you can take a look to get a feel for the structure, but it doesn't need a detailed review. `pyproject.toml` is now the home for the

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,7 +29,27 @@ $ docker build -t harvard-atg/media_management_api .
...
 ```
 
-**Other tasks:**
+### Setting up dev environment
+
+**Dependencies**
+
+The requirements are stored in `pyproject.toml`, and `poetry.lock` contains the actual dependencies as installed in the virtual environment that can be used for development. Poetry installation instructions are available in [Poetry documentation](https://python-poetry.org/docs/).
+
+With poetry installed, you can specify a python execeutable with `poetry env use /path/to/python`, so it plays nice with [pyenv](https://github.com/pyenv/pyenv). Then a `poetry install` will install dependencies and `poetry shell` will get you into the environment. Add dependencies with `poetry add packagename`, or `poetry add --dev packagename` for dev dependencies. Check out [Poetry documentation](https://python-poetry.org/docs/) for a full overview.
```
Comment (elliottyates) [inline README.md:38]: nit: typo: executable

**Review by elliottyates — COMMENTED** [review]
This is great, Jeremy -- nice addition of poetry, and docs for it, thanks for letting us know how easy it is to set up! (At least, you made it look easy. 🙂 )

**Comment by elliottyates** [comment]
One additional comment might be that in future it can be helpful for all config to be in its own set of commits, and all actual linting in another (e.g. https://github.com/Harvard-University-iCommons/exed_user_tool/pull/26/commits). That can make it easier for reviewers.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        working-directory: .
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Install deps using Pip
+        run: pip install isort black flake8
```
[the remark this answers is not in the document]
Comment (jaguillette) [reply .github/workflows/ci.yml:27]: That'd be a cool thing to have in place, although I think I'd prefer to go with the pre-commit.ci route in that case, with that action being in a maintenance-only mode. I'll keep it in mind if maintenance gets too cumbersome. Thanks!

File: README.md [hunk 3]
```diff
@@ -29,7 +29,27 @@ $ docker build -t harvard-atg/media_management_api .
 $ docker tag harvard-atg/media_management_api:latest harvard-atg/media_management_api:dev
 ```
 
-**Other tasks:**
+### Setting up dev environment
+
+**Dependencies**
+
+The requirements are stored in `pyproject.toml`, and `poetry.lock` contains the actual dependencies as installed in the virtual environment that can be used for development. Poetry installation instructions are available in [Poetry documentation](https://python-poetry.org/docs/).
```
[the remark this answers is not in the document]
Comment (jaguillette) [reply README.md:36]: Exactly, it's got all the versions of all the dependencies in that lock file. It's also got the exact versions on the off chance a minor or patch version change in a top-level specification with fuzzy matching causes a problem.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        if: steps.python-cache.outputs.cache-hit != 'true'
```
[the remark this answers is not in the document]
Comment (jaguillette) [reply .github/workflows/ci.yml:29]: Added! I had to figure it out for myself, having copied this over from IIIFingest, but it's a caching step: https://github.com/actions/cache

File: README.md (hunk 1 above)
Comment (jaguillette) [reply README.md]: fixed, thanks!

**Review by arthurian — APPROVED** [review]
This all looks good to me @jaguillette. 

One thing to keep in mind is that the ansible playbooks that deploy to EC2 look for a `requirements.txt` file (see [django_project_deploy](https://github.com/Harvard-University-iCommons/tlt-ops/blob/fd1721e033da152c02543de1f2a0988aafb0319c/ansible/roles/django_project_deploy/tasks/main.yml#L76)). So until we've fully migrated to ECS Fargate, we might want to keep one around, using poetry as the source of truth. Looks like there's an [export](https://python-poetry.org/docs/cli/#export) command that could help with this.

**Comment by jaguillette** [comment]
Thanks! I added a pre-commit hook to make sure that a requirements.txt gets generated from the poetry environment and that it's up to date. A github action to do that even more automatically in the future would be a great addition!

**Pull request merged**